### PR TITLE
feat: add kubernetes_api_service for new pipeline

### LINKED ISSUE #340: Create the job resource using the API

### PR DESCRIPTION
**Pull request by shiipou**
Signed-off-by: shiipou <shiishii@nocturlab.fr>
Closes #340 

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [ ] I included unit tests that cover my changes
- [x] I added/updated the documentation about my changes
- [x] I made my own code-review before requesting one

### CODE REVIEW

**Review by taorepoara — APPROVED** [review]
Seems OK

@pichoemr Can you look at it two please

**Comment by taorepoara** [comment]
@shiipou You have a formating issue in the CI

File: apps/lenra/lib/lenra/services/kubernetes_api_services.ex [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    body =
+      Jason.encode!(%{
+        apiVersion: "batch/v1",
+        kind: "Job",
+        metadata: %{
+          name: build_name,
+          namespace: kubernetes_build_namespace
+        },
+        spec: %{
```
Comment (taorepoara) [inline apps/lenra/lib/lenra/services/kubernetes_api_services.ex:68]: The job does not have resources management

File: apps/lenra/lib/lenra/apps.ex [hunk 2]
```diff
@@ -214,13 +214,25 @@ defmodule Lenra.Apps do
...
+                build.build_number
+              )
+          "kubernetes" -> KubernetesApiServices.create_pipeline(
+              app.service_name,
+              app.repository,
+              app.repository_branch,
+              build.id,
+              build.build_number
+            )
+          _ -> BusinessError.pipeline_runner_unkown_service()
```
Comment (pichoemr) [inline apps/lenra/lib/lenra/apps.ex:233]: You should return a tuple `{:error, <error>}` inside a run:

```suggestion
          _ -> BusinessError.pipeline_runner_unkown_service_tuple()
```

File: config/releases.exs [hunk 3]
```diff
@@ -27,7 +27,14 @@ config :lenra,
   gitlab_ci_ref: "master",
   template_url: System.fetch_env!("TEMPLATE_URL"),
   lenra_email: System.fetch_env!("LENRA_EMAIL"),
-  lenra_app_url: System.fetch_env!("LENRA_APP_URL")
+  lenra_app_url: System.fetch_env!("LENRA_APP_URL"),
+  pipeline_runner: System.get_env("PIPELINE_RUNNER", "gitlab"),
+  kubernetes_api_url: System.get_env("KUBERNETES_API_URL") || "https://#{System.get_env("KUBERNETES_SERVICE_HOST")}",
```
Comment (pichoemr) [inline config/releases.exs:32]: In the case KUBERNETES_API_URL & KUBERNETES_SERVICE_HOST are not set you will have an url: `"https://"`. Maybe only let KUBERNETES_API_URL with `fetch_env!`

**Review by pichoemr — CHANGES_REQUESTED** [review]

**Review by shiipou — COMMENTED** [review]
fix: use fetch_env to not allow empty var `KUBERNETES_SERVICE_HOST` if `KUBERNETES_API_URL` is not defined

File: apps/lenra/lib/lenra/services/kubernetes_api_services.ex [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+              }]
+            },
+            resources: %{
+              "limits": %{
+                "cpu":"500m",
+                "memory":"1024Mi"
+              },
+              "requests": %{
+                "cpu":"50m",
+                "memory":"100Mi"
```
Comment (pichoemr) [inline apps/lenra/lib/lenra/services/kubernetes_api_services.ex:159]: Missing spaces after `:` that cause compile error
```suggestion
                "cpu": "500m",
                "memory": "1024Mi"
              },
              "requests": %{
                "cpu": "50m",
                "memory": "100Mi"
```

**Review by pichoemr — CHANGES_REQUESTED** [review]

**Review by pichoemr — APPROVED** [review]
Fix CI with a `mix format` and it's good for me

**Review by pichoemr — CHANGES_REQUESTED** [review]

File: config/dev.exs [hunk 5]
```diff
@@ -139,7 +139,14 @@ config :lenra,
   faas_secrets: ["gitlab-registry"],
   template_url: System.get_env("TEMPLATE_URL", "https://github.com/lenra-io/templates.git#beta"),
   lenra_email: System.get_env("LENRA_EMAIL", "contact@lenra.io"),
-  lenra_app_url: System.get_env("LENRA_APP_URL", "https://localhost:10000")
+  lenra_app_url: System.get_env("LENRA_APP_URL", "https://localhost:10000"),
+  pipeline_runner: System.get_env("PIPELINE_RUNNER", "GitLab"),
```
[the remark this answers is not in the document]
Comment (shiipou) [reply config/dev.exs:143]: Why that ?
GitLab or gitlab or gitLab is the same because the check does a toLowerCase in the code.

**Comment by shiipou** [comment]
I need to document this addition.

File: docs/env-variables.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
+# Environment Variables
+
+This document provides a list of the environment variables that need to be set for the application to function correctly. Default values are provided where applicable.
+
+## General Configuration
+
+- `PORT`: The port on which the application will run.
+- `SECRET_KEY_BASE`: The secret key base for the application.
+- `API_ENDPOINT`: The endpoint for the application's API.
+- `APP_HOST`: The host of the application.
```
Comment (taorepoara) [inline docs/env-variables.md:10]:
```suggestion
- `APP_HOST`: The host of the API. This variable should change in the futur.
```

**Review by taorepoara — CHANGES_REQUESTED** [review]

**Review by taorepoara — APPROVED** [review]
Seems OK

File: docs/env-variables.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+- `KUBERNETES_API_TOKEN`: The token for the Kubernetes API. If not set, the value is read from the path specified in `KUBERNETES_API_TOKEN_PATH`.
+- `KUBERNETES_API_TOKEN_PATH`: The path to the Kubernetes API token. Default is "/var/run/secrets/kubernetes.io/serviceaccount/token".
+- `KUBERNETES_BUILD_NAMESPACE`: The namespace for Kubernetes builds. Default is "lenra-build".
+- `KUBERNETES_BUILD_SCRIPTS`: The scripts for Kubernetes builds. Default is "lenra-build-scripts".
+- `KUBERNETES_BUILD_SECRET`: The secret for Kubernetes builds. Default is "lenra-build-secret".
+
+## Template Configuration
+
+- `TEMPLATE_URL`: The URL for the template.
+
```
Comment (shiipou) [inline docs/env-variables.md:50]:
```suggestion
```

**Pull request merged**